### DISCUSSION TITLE
tooltips for displaying footnotes

### DISCUSSION THREAD

**Discussion by diomed** [satnaing/astro-paper#219] [Ideas]
would be great if theme had option for tooltips to display footnotes

**Comment by tcn33** [comment]
I've added [Littlefoot](https://github.com/goblindegook/littlefoot) to my site, but just need to work out one small problem.

I've made the following changes:
- Downloaded `littlefoot.css` from the repo and put it inside `src/styles/littlefoot.css`
- Added `import "@styles/littlefoot.css` to the header of `Layout.astro`
- Updated the bottom of `Layout.astro`:
```javascript
  </head>
  <body>
    <slot />
    <script is:inline
      src="https://unpkg.com/littlefoot/dist/littlefoot.js"
      type="application/javascript">
    </script>
    <script is:inline type="application/javascript">
      littlefoot.littlefoot();
    </script>
  </body>
</html>
```
Result: 
![CleanShot 2024-11-05 at 17 36 43@2x](https://github.com/user-attachments/assets/6bbc207f-1cba-4b47-94ee-7bb72e928efc)
... but only if I hit reload on that page. If I initially load any other page, or navigate away from the first page and then back, it just displays the normal footnote. Just need to figure out where I should be calling that `littlefoot.littlefoot()` function instead.

edit: Fixed it for now.
```javascript
 <script>
   document.addEventListener('astro:page-load', () => {
     littlefoot.littlefoot();
   });
 </script>
```

Side note, I'm sure I'm doing this really inefficiently - I just couldn't figure out a way to add it to `astro.config.ts` as it kept throwing "Document is not defined" errors.
